### PR TITLE
libs: remove useless code in group

### PR DESCRIPTION
**Pull request by guagualvcha**
I am reading code about wal code, auto file package contains some confusing code that is useless

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by Liamsi** [comment]
Added the breaking label as this breaks the go-API. Deleting a bunch of code, my favourite PRs 👍

**Comment by guagualvcha** [comment]
Thank you for quick review.

**Comment by guagualvcha** [comment]
> @guagualvcha Great change! Deleting code like no tomorrow.
> 
> Could you update the changelog?

sure

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 * Apps
 
 * Go API
+- [libs/autofile] Remove useless code in autofile package
```
Comment (Liamsi) [inline CHANGELOG_PENDING.md]:
```suggestion
- [libs/autofile] \#3504 Remove useless code in autofile package (@guagualvcha)
```

Comment (melekes) [reply CHANGELOG_PENDING.md]: Would also be good to enumerate the removed public functions, namely: `Group#Search, Group#FindLast, GroupReader#ReadLine, GroupReader#PushLine`

Comment (Liamsi) [reply CHANGELOG_PENDING.md]: Very good point. I've updated the suggestion above.

**Pull request merged**